src/main/java/net/minecraftforge/event/level/SaplingGrowTreeEvent.java

```diff
@@ -6,38 +6,52 @@
 package net.minecraftforge.event.level;
 
 import net.minecraft.core.BlockPos;
+import net.minecraft.core.Holder;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.util.RandomSource;
 import net.minecraft.world.level.LevelAccessor;
 import net.minecraft.world.level.block.SaplingBlock;
 import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.levelgen.feature.ConfiguredFeature;
 import net.minecraftforge.common.MinecraftForge;
 import net.minecraftforge.eventbus.api.Cancelable;
 import net.minecraftforge.eventbus.api.Event.HasResult;
 
 /**
- * This event is fired whenever a sapling grows into a tree.
+ * This event is fired whenever a sapling, fungus, mushroom or azalea grows into a tree.
  * This event is fired during sapling growth in
  * {@link SaplingBlock#advanceTree(ServerLevel, BlockPos, BlockState, RandomSource)}.
  * <p>
  * This event is not {@linkplain Cancelable cancellable} but does {@linkplain HasResult have a result}.
- * {@linkplain Result#ALLOW ALLOW} and {@linkplain Result#DEFAULT DEFAULT} will allow the sapling to grow.
+ * {@linkplain Result#ALLOW ALLOW} and {@linkplain Result#DEFAULT DEFAULT} will allow the sapling to grow
+ * using the features set on the event.
  * {@linkplain Result#DENY DENY} will prevent the sapling from growing.
  * <p>
  * This event is fired on the {@linkplain MinecraftForge#EVENT_BUS main Forge event bus}
  * only on the {@linkplain net.minecraftforge.fml.LogicalSide#SERVER logical server}.
+ * TODO 1.20 Change to cancelable and rename event
```

Comment:
I think the todo should only mention renaming. Once 1.20 comes around, we can reevaluate whether to change this or not.